Source/Orts.Simulation/Common/Scripting/ScriptManager.cs

```diff
@@ -47,6 +47,7 @@ static CompilerParameters GetCompilerParameters()
             cp.ReferencedAssemblies.Add("System.dll");
             cp.ReferencedAssemblies.Add("System.Core.dll");
             cp.ReferencedAssemblies.Add("ORTS.Common.dll");
+            cp.ReferencedAssemblies.Add("Orts.Formats.Msts.dll");
```

Comment:
I'm not happy with this, I would prefer converting MstsSignalAspect and MstsBlockState to other enum types so this is not required. Good candidates are Aspect enum as used in TrainControlSystem and InternalBlockState used by signalling logic, or new enums can be created.

Reply:
I think you can use Aspect from ORTS.Scripting.Api without any trouble. You just have to convert it back at the right locations inside OR.